components/notification-sender-config/org.wso2.carbon.identity.notification.sender.tenant.config/src/main/java/org/wso2/carbon/identity/notification/sender/tenant/config/NotificationSenderManagementServiceImpl.java

```diff
@@ -551,4 +584,55 @@ private String getChannelTypeFromResource(Resource resource) {
                     .map(Attribute::getValue).orElse(DEFAULT_HANDLER_NAME);
         }
     }
+
+    /**
+     * Method to check whether the sender is allowed to delete.
+     *
+     * @param senderName    Name of the sender.
+     * @return  whether the sender is allowed to delete.
+     * @throws NotificationSenderManagementException If an error occurred while checking the sender.
+     */
+    private boolean checkAllowDelete(String senderName) throws NotificationSenderManagementException {
+
+        if (SENDERS.get(senderName) != null) {
+            String authenticatorId = new String(Base64.getUrlEncoder()
+                    .encode(SENDERS.get(senderName).getBytes(StandardCharsets.UTF_8)),
+                    StandardCharsets.UTF_8);
+            String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            try {
+                ConnectedAppsResult appsResult =
+                        NotificationSenderTenantConfigDataHolder.getInstance()
+                        .getApplicationManagementService()
+                        .getConnectedAppsForLocalAuthenticator(authenticatorId, tenantDomain, 1 , 0);
+                // If there are any connected apps, the sender cannot be deleted.
+                if (appsResult.getApps().size() > 0) {
+                    return false;
+                }
+                Resource resource = NotificationSenderTenantConfigDataHolder.getInstance().getConfigurationManager()
+                        .getResource(MY_ACCOUNT_SMS_RESOURCE_TYPE, MY_ACCOUNT_SMS_RESOURCE_NAME);
+                String smsOtpEnabled = resource.getAttributes().stream().filter(attribute -> attribute
+                        .getKey().equals("sms_otp_enabled")).collect(
+                        Collectors.toList()).get(0).getValue();
+                // If SMS OTP is enabled for my account, the sender cannot be deleted.
+                if (Boolean.parseBoolean(smsOtpEnabled)) {
+                    return false;
+                }
+            } catch (IdentityApplicationManagementException e) {
+                if (e instanceof IdentityApplicationManagementClientException && e.getErrorCode()
+                        .equals(Error.AUTHENTICATOR_NOT_FOUND.getCode())) {
```

Comment:
Swap the constant and the key word like ((Error.AUTHENTICATOR_NOT_FOUND.getCode()).equals(e.getErrorCode()) to avoid possible null pointers.

Reply:
Addressed with 15a1ae7633cf62024d90506b9322d1cea1494c3c